ddl/column.go

```diff
@@ -1102,6 +1099,120 @@ func (w *worker) doModifyColumnTypeWithData(
 	return ver, errors.Trace(err)
 }
 
+func replaceOldColumn(tblInfo *model.TableInfo, oldID, newID int64,
+	newName model.CIStr) (changingCol *model.ColumnInfo, err error) {
+	// Locate the old column and new column.
+	oldCol := model.FindColumnInfoByID(tblInfo.Columns, oldID)
+	if oldCol == nil {
+		return nil, infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
+	}
+	changingCol = model.FindColumnInfoByID(tblInfo.Columns, newID)
+	if changingCol == nil {
+		return nil, infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
+	}
+	// Replace the old column.
+	tblInfo.Columns[oldCol.Offset] = changingCol
+	tblInfo.Columns[changingCol.Offset] = nil
+	changingCol = updateChangingCol(changingCol, newName, oldCol.Offset)
+	// Remove nil column.
+	tmp := tblInfo.Columns[:0]
+	for _, c := range tblInfo.Columns {
+		if c != nil {
+			tmp = append(tmp, c)
+		}
+	}
+	tblInfo.Columns = tmp
+	return changingCol, nil
+}
+
+func replaceOldIndexes(tblInfo *model.TableInfo, changingIdxs []*model.IndexInfo) {
+	// Remove the changing indexes.
+	for i, idx := range tblInfo.Indices {
+		for _, cIdx := range changingIdxs {
+			if cIdx.ID == idx.ID {
+				tblInfo.Indices[i] = nil
+				break
+			}
+		}
+	}
+	tmp := tblInfo.Indices[:0]
+	for _, idx := range tblInfo.Indices {
+		if idx != nil {
+			tmp = append(tmp, idx)
+		}
+	}
+	tblInfo.Indices = tmp
+	// Replace the old indexes with changing indexes.
+	for _, cIdx := range changingIdxs {
+		idxName := getChangingIndexOriginName(cIdx)
+		for i, idx := range tblInfo.Indices {
+			if strings.EqualFold(idxName, idx.Name.O) {
+				cIdx.Name = model.NewCIStr(idxName)
```

Comment:
Please add a comment on why add this. I almost forgot why I did it

Reply:
I have add the comment to line 1148.